unic/ucd/ident/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2017 The UNIC Project Developers.
+//
+// See the COPYRIGHT file at the top-level directory of this distribution.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+#![forbid(unsafe_code, unconditional_recursion, missing_docs)]
+
+//! # UNIC — UCD — Identifier Properties
+//!
+//! A component of [`unic`: Unicode and Internationalization Crates for Rust](/unic/).
+//!
+//! Accessor to the UCD properties used widely by [UAX31 Unicode Identifier and Pattern Syntax].
+//!
+//! # Features
+//!
+//! - `xid` (default): the `XID_Start` and `XID_Continue` properties
+//! - `id` (optional): the `ID_Start` and `ID_Continue` properties;
+//!   _in most cases, you should prefer using the `XID` properties
+//!   because they are consistent under NFKC normalization_
+//! - `pattern` (optional): the `Pattern_Syntax` and `Pattern_White_Space` properties
+//!
+//! [UAX31 Unicode Identifier and Pattern Syntax]: <http://www.unicode.org/reports/tr31/>
+
+#[macro_use]
+extern crate unic_char_property;
+#[macro_use]
+extern crate unic_char_range;
+extern crate unic_ucd_core;
+
+
+#[cfg(feature = "xid")]
+mod xid {
+    char_property! {
+        /// A character that can start an identifier, stable under NFKC.
+        pub struct XidStart(bool) {
+            abbr => "XIDS";
+            long => "XID_Start";
+            human => "NFKC Identifier Start";
```

Comment:
I think we better stick as close to the `long` format as possible, and only expand abbreviations, if there's any. Using "NFKC Identifier" for "XID" here is not common at all and I think we better keep it as simple as "XID Start".

Basically, how I look at the `human` value is "what we expect to see in a table header", where we don't want to stay machine-readable or have underscore joins, and there's room to not over-abbreviate.

Reply:
ID Start is obviously an "Identifier Start". XID Start is ID Start modified for closure under NFKC. Thus, "NFKC Identifier Start".

But yes, just using human => "XID Start" would be less confusing.